ui/src/app/views/notebook-redirect/component.spec.ts

```diff
@@ -105,140 +104,234 @@ describe('NotebookRedirectComponent', () => {
   }));
 
-  it('should display "Redirecting" while redirecting', fakeAsync(() => {
-    blockingClusterStub.cluster.status = ClusterStatus.Running;
+  it('should display spinners to match progress status when creating a notebook', fakeAsync(() => {
+    spyOn(window, 'setTimeout').and.stub();
+    updateAndTick(fixture);
+    fixture.componentInstance.progress = fixture.componentInstance.Progress.Initializing;
+    updateAndTick(fixture);
+    tick(1000);
+    expect(fixture.debugElement.queryAll(By.css('#Initializing'))[0]
+        .children[0].nativeElement).toBe(spinner());
+
+    fixture.componentInstance.progress = fixture.componentInstance.Progress.Authenticating;
+    updateAndTick(fixture);
+    tick(1000);
+    expect(fixture.debugElement.queryAll(By.css('#Authenticating'))[0]
+        .children[0].nativeElement).toBe(spinner());
+
+    fixture.componentInstance.progress = fixture.componentInstance.Progress.Creating;
+    updateAndTick(fixture);
+    tick(1000);
+    expect(fixture.debugElement.queryAll(By.css('#Creating'))[0]
+        .children[0].nativeElement).toBe(spinner());
+
+    fixture.componentInstance.progress = fixture.componentInstance.Progress.Redirecting;
+    updateAndTick(fixture);
+    tick(10000);
+    expect(fixture.debugElement.queryAll(By.css('#Redirecting'))[0]
+        .children[0].nativeElement).toBe(spinner());
+
+  }));
+
+  it('should display spinners to match progress status when loading a notebook', fakeAsync(() => {
+    spyOn(window, 'setTimeout').and.stub();
     updateAndTick(fixture);
+    fixture.componentInstance.notebookName = 'foo.ipynb';
+    fixture.componentInstance.creating = false;
     fixture.detectChanges();
-    expect(spinnerText()).toContain('Redirecting');
+    fixture.componentInstance.progress = fixture.componentInstance.Progress.Resuming;
+    updateAndTick(fixture);
+    expect(fixture.debugElement.queryAll(By.css('#Initializing'))[0]
+        .children[0].nativeElement).toBe(spinner());
+
+    fixture.componentInstance.progress = fixture.componentInstance.Progress.Authenticating;
+    updateAndTick(fixture);
+    expect(fixture.debugElement.queryAll(By.css('#Authenticating'))[0]
+        .children[0].nativeElement).toBe(spinner());
+
+    fixture.componentInstance.progress = fixture.componentInstance.Progress.Copying;
+    updateAndTick(fixture);
+    expect(fixture.debugElement.queryAll(By.css('#Copying'))[0]
+        .children[0].nativeElement).toBe(spinner());
+
+    fixture.componentInstance.progress = fixture.componentInstance.Progress.Redirecting;
+    updateAndTick(fixture);
+    expect(fixture.debugElement.queryAll(By.css('#Redirecting'))[0]
+        .children[0].nativeElement).toBe(spinner());
+
   }));
 
+  it('should escape notebook names', fakeAsync(() => {
+    blockingClusterStub.cluster.status = ClusterStatus.Running;
+    updateAndTick(fixture);
+    fixture.detectChanges();
+    tick(10000);
+    expect(fixture.componentInstance.leoUrl).toMatch('blah%20blah');
```

Comment:
This still isn't reproducing the original bug. The initial value needs to contain an escape sequence. Please just use my original test value which was taken from an actual bug we saw: `1%2B1.ipynb`

Also, is it not possible to inspect the iframe URL? THat would be preferred to digging into the component properties.

Reply:
Ok. So in manual testing if I set a new notebook to the name '1%2B1.ipynb'. A new notebook is created with the name '1+1.ipynb' and our url shows '/1%252B1.ipynb', the iframe source shows '1+1.ipynb', and the breadcrumb shows '1%2B1.ipynb'.... The only issue I see with any of that is possibly the breadcrumb. I'm going to update the test to look at the iframe src but expect '1+1.ipynb'.

Reply:
OK, that doesn't look to be reproducing the conditions of the bug. I think the behavior of the new notebook modal is probably fine, but to see the bug you need to have a notebook that actually has a name (in GCS) with a URL encoded escape sequence in it, i.e. your actual notebook should have name `1%2B1.ipynb`.

You can probably get there by specifying `1%252B1.ipynb` as the new notebook name in Workbench.

It's an edge case, but we have actually hit this bug before. Likely originally it was generated by uploading a file into Jupyter, or creating/renaming via the Jupyter UI itself.